include/onnxruntime/core/framework/allocator.h

```diff
@@ -25,6 +25,8 @@ constexpr const char* CUDA_PINNED = "CudaPinned";
 constexpr const char* MIGRAPHX = "MIGraphX";
 constexpr const char* MIGRAPHX_PINNED = "MIGraphXPinned";
 
+constexpr size_t kAllocAlignment = 256;
```

Comment:
Why choose 256 over 64? We were already using 64 for activations and I understand 256 may be preferred for CUDA but my concern here is you can potentially increase the footprint by padding with more bytes. At the very least do some benchmarks to measure before/after memory footprint across few workloads.

Reply:
The largest potential waste for one tensor is 255 bytes. Let's say we have 10,000 tensor (as a reference, RobertXXXL, a 2Billion parameter model has ~2000 gradients), and we are extremely unlucky that all the tensor need padding, we will waste ~2.5MB on the padding space. 

In real world, we will most of the tensors comes in the hidden dimension of 128, 256, 768, 1024, 2048 etc. 
So I am not too concerned about the increase in memory usage.

Reply:
Sometimes it's that 2.5mb that makes the difference between one batch size and the next. Also DNN models are getting bigger exponentially so expect the same with wastage. 

Why not settle with 64? How much perf difference is there between 64 and 256 on CUDA?

Reply:
It has something to do with GPU cacheline size, see [https://forums.developer.nvidia.com/t/pascal-l1-cache/49571](https://forums.developer.nvidia.com/t/pascal-l1-cache/49571). Since cudaMalloc always return address that's 256-aligned, it's better to be consistent to that to ensure we have the same cacheline perf with different allocators.

---
In reply to: [540756443](https://github.com/microsoft/onnxruntime/pull/6109#discussion_r540756443) [](ancestors = 540756443)